src/scheduler/fifo.c

```diff
@@ -2327,3 +2350,358 @@ next_job(status *policy, server_info *sinfo, int flag)
 	}
 	return rjob;
 }
+
+/**
+ * @brief
+ *	Helper function used to copy the given source string to the destination string. It also
+ *	frees the destination and then allocates required memory before copying.
+ *
+ * @param[in] dest - Address of pointer to destination string
+ * @param[in] dest - pointer to source string
+ *
+ * @retval
+ * @return 0 - Failure
+ * @return  1 - Success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+static int
+copy_attr_value(char **dest, char *src)
+{
+	int len = 0;
+	int ret = 0;
+
+	if (*dest != NULL)
+		free(*dest);
+
+	if (src !=NULL) {
+		len = strlen(src);
+		*dest = (char*)malloc(len + 1);
+		if (*dest == NULL) {
+			log_err(errno, __func__, MEM_ERR_MSG);
+			return ret;
+		}
+		strncpy(*dest, src, len);
+		(*dest)[len] = '\0';
+		ret = 1;
+	}
+	return ret;
+}
+
+/**
+ * @brief
+ *	Helper function used to copy the attribute values from batch_status to the corresponding
+ *	scheduler global variables which hold its priv_dir, log_dir and partitions
+ *
+ * @param[in] status - populated batch_status after stating this scheduler from server
+ *
+ * @retval
+ * @return 0 - Failure
+ * @return  1 - Success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+static int
+sched_settings_frm_svr(struct batch_status *status)
+{
+	struct attrl *attr;
+	char *tmp_priv_dir = NULL;
+	char *tmp_log_dir = NULL;
+	char *tmp_partitions = NULL;
+	struct	attropl	*attribs;
+	struct batch_status *ss = NULL;
+	static char *log_dir = NULL;
+	char *tmp_comment = NULL;
+	int clear_comment = 0;
+	extern char *partitions;
+
+	attr = status->attribs;
+
+	 /* resetting the following before fetching from batch_status. */
+	while (attr != NULL) {
+		if (attr->name != NULL && attr->value != NULL) {
+			if (!strcmp(attr->name, ATTR_sched_priv)) {
+				if (copy_attr_value(&tmp_priv_dir, attr->value) == 0)
+					return 0;
+			} else if (!strcmp(attr->name, ATTR_sched_log)) {
+				if (copy_attr_value(&tmp_log_dir, attr->value) == 0)
+					return 0;
+			} else if (!strcmp(attr->name, ATTR_partition)) {
+				if (copy_attr_value(&tmp_partitions, attr->value) == 0)
+					return 0;
+			} else if (!strcmp(attr->name, ATTR_comment)) {
```

Comment:
I'm not sure if I have asked this before, but why do we need a comment to be stored in scheduler?